tests/jballerina-unit-test/src/test/resources/test-src/types/stream/stream-negative.bal

```diff
@@ -380,3 +380,11 @@ function testAssignabilityOfStreams() {
 function testNonIteratorsInStreamConstructor() {
     stream<int> intStrm = new("test");
 }
+
+function testUnboundedStreamWithEmptyConstructor() {
+    stream<int, error?> _ = new;
+    stream<int, error?> _ = new stream<int, error?>();
+    stream<int, readonly> _ = new stream<int, readonly>();
+    var _ = new stream<int, error?>();
```

Comment:
These are positive tests? Shall we move them to the respective file?
But `stream<int, readonly> _ = new stream<int, readonly>();` causes a TypeCastError at runtime?

Shall we also add some positive tests where the stream constructor is provided for the unbounded streams?

Reply:
Added the tests. readonly assignability has an issue https://github.com/ballerina-platform/ballerina-lang/issues/35847. Will be restricted at compile time after fixing it